From 9edda7cae4d5a9522b87b0035705d9dbb3f7502e Mon Sep 17 00:00:00 2001
From: sbruno <sbruno@FreeBSD.org>
Date: Mon, 21 May 2012 22:23:21 +0000
Subject: [PATCH 2/9] MFC r235634

Fix and update battery status bits according to linux driver
---
 sys/dev/mfi/mfireg.h |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/sys/dev/mfi/mfireg.h b/sys/dev/mfi/mfireg.h
index 5bed6c2..dab9cf7 100644
--- a/sys/dev/mfi/mfireg.h
+++ b/sys/dev/mfi/mfireg.h
@@ -1377,8 +1377,13 @@ struct mfi_bbu_status {
 #define	MFI_BBU_STATE_PACK_MISSING	(1 << 0)
 #define	MFI_BBU_STATE_VOLTAGE_LOW	(1 << 1)
 #define	MFI_BBU_STATE_TEMPERATURE_HIGH	(1 << 2)
-#define	MFI_BBU_STATE_CHARGE_ACTIVE	(1 << 0)
-#define	MFI_BBU_STATE_DISCHARGE_ACTIVE	(1 << 0)
+#define	MFI_BBU_STATE_CHARGE_ACTIVE	(1 << 3)
+#define	MFI_BBU_STATE_DISCHARGE_ACTIVE	(1 << 4)
+#define	MFI_BBU_STATE_LEARN_CYC_REQ	(1 << 5)
+#define	MFI_BBU_STATE_LEARN_CYC_ACTIVE	(1 << 6)
+#define	MFI_BBU_STATE_LEARN_CYC_FAIL	(1 << 7)
+#define	MFI_BBU_STATE_LEARN_CYC_TIMEOUT	(1 << 8)
+#define	MFI_BBU_STATE_I2C_ERR_DETECT	(1 << 9)
 	uint8_t			pad[20];
 	union mfi_bbu_status_detail detail;
 } __packed;
-- 
1.7.10.4

